quickwit-ui/app/src/utils/models.ts

```diff
@@ -0,0 +1,215 @@
+    }
+  }
+  return value;
+}
+
+export function flattenEntries(doc_mapping: DocMapping, raw_doc: RawDoc): Entry[] {
+  const flatten_fields = getFlattenFields(doc_mapping.field_mappings);
+  const records = [];
+
+  for (let flatten_field of flatten_fields) {
+    const value = getValueFromPath(flatten_field.path, raw_doc);
+    if (value !== null) {
+      records.push({key: flatten_field.name, value: value});
+    }
+  }
+  return records;
+}
+
+function getFlattenFields(field_mappings: FieldMapping[]): FlattenField[] {
+  let fields: FlattenField[] = [];
+  for (let field_mapping of field_mappings) {
+    if (field_mapping.type === 'object' && field_mapping.field_mappings !== undefined) {
+      for (let child_field of getFlattenFields(field_mapping.field_mappings)) {
+        fields.push({name: field_mapping.name + '.' + child_field.name, path: [field_mapping.name].concat(child_field.path), type: child_field.type})
+      }
+    } else {
+      fields.push({name: field_mapping.name, path: [field_mapping.name], type: field_mapping.type});
+    }
+  }
+   
+  return fields;
+}
+
+export function guessTimeUnit(index: Index): TimeUnit {
+  if (index.splits.length === 0 || index.metadata.indexing_settings.timestamp_field === null) {
+    return TimeUnit.MILLI_SECOND;
+  }
+  // Not possible in theory as guessTimeUnit function is 
+  // called only on index with a timestamp field.
+  if (index.splits[0].time_range === null) {
+    return TimeUnit.MILLI_SECOND;
+  }
+  let range_start_values = index.splits.map(split => split.time_range === null ? 0 : split.time_range.start);
+  const time_range_start_max = Math.max(...range_start_values);
+  // We expect a split time range to be between year between 1971 and 2070. 
+  const seconds_in_one_hundred_years = 3600 * 24 * 365 * 100;
+  if (time_range_start_max < seconds_in_one_hundred_years) {
+    return TimeUnit.SECOND
+  }
+  if (time_range_start_max < seconds_in_one_hundred_years * 1000) {
+    return TimeUnit.MILLI_SECOND
+  }
+  if (time_range_start_max < seconds_in_one_hundred_years * 1000 * 1000) {
+    return TimeUnit.MICRO_SECOND
+  }
+  console.warn('Cannot guess correctly time unit, value `time_range_start_max` is too high, set to micro seconds', time_range_start_max);
+  return TimeUnit.MICRO_SECOND
+}
+
+export function getAllFields(doc_mapping: DocMapping) {
+  return getFlattenFields(doc_mapping.field_mappings);
+} 
+
+export type DocMapping = {
+  field_mappings: FieldMapping[];
+  tag_fields: string[];
+  store: boolean;
+}
+
+export type SearchRequest = {
+  indexId: string | null;
+  query: string;
+  startTimestamp: number | null;
+  endTimestamp: number | null;
+  numHits: number;
+}
+
+export const EMPTY_SEARCH_REQUEST: SearchRequest = {
+  indexId: '',
+  query: '',
+  startTimestamp: null,
+  endTimestamp: null,
+  numHits: 100,
+}
+
+export type SearchResponse = {
+  count: number;
+  hits: Array<RawDoc>;
+  numMicrosecs: number;
+}
+
+export type IndexMetadata = {
```

Comment:
we should consider generating these types

https://github.com/Aleph-Alpha/ts-rs

Reply:
oh nice! Will definitely give it a try. I will put that in a separate issue.